ocaml/xenopsd/xc/xenops_server_xen.ml

```diff
@@ -1121,6 +1121,18 @@ module HOST = struct
               features
         )
     )
+
+  let combine_cpu_policies policy1 policy2 =
+    let open Cpuid in
+    intersect (features_of_string policy1) (features_of_string policy2)
+    |> string_of_features
+
+  let is_compatible vm_policy host_policy =
+    let open Cpuid in
+    let vm = features_of_string vm_policy in
+    let host = features_of_string host_policy in
+    let vm' = zero_extend vm (Array.length host) in
+    is_subset vm' host
```

Comment:
See the XAPI code, it would be useful to print a diff here, that helps determining what went wrong in many cases.

Reply:
Yes, we compute and can print the diff here now. Once we switch to the C version of the function to do the comparison, then we would need that function to return a diff in some form.

Reply:
I've done that now.